integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonDropDataMapCommand.scala

```diff
@@ -114,12 +114,6 @@ case class CarbonDropDataMapCommand(
         LOGGER.audit(s"Deleting datamap [$dataMapName] under table [$tableName]")
 
         // drop index datamap on the main table
-        if (mainTable != null &&
```

Comment:
Can you explain why this modification is needed?

Reply:
when drop preagg datamap  , should delete the datamap schema file and main tableinfo schema info.